gilded-rose-kata/src/main/kotlin/models/GildedRoseQualityFactory.kt

```diff
@@ -0,0 +1,28 @@
+package models
+
+import enums.*
+import interfaces.*
+
+class GildedRoseQualityFactory(override var quality: Int) : QualityFactory {
+
+    companion object {
+        const val MINIMUM_THRESHOLD = 0
```

Comment:
Looks like those should be private. I'd recommend you to move them to the top level above the class definition to avoid the creation of an unused companion object (keep in mind each companion object is a static instance in memory).

Reply:
I am not 100% sure about the implications of having a static instance in memory. The main concern relies on this being a practice and, when it turns to be a big project, multiple static instances in memory that could be avoided make a project be heavier to the memory than it should?